specifyweb/frontend/js_src/lib/components/TreeView/index.tsx

```diff
@@ -274,12 +264,40 @@ export function TreeViewWrapper(): JSX.Element | null {
   );
 }
 
-function useTreeStates() {
+function useStates<SCHEMA extends AnyTree>(
+  tableName: SCHEMA['tableName']
+): {
+  readonly focusedRow: Row | undefined;
```

Comment:
I was going to use a GetOrSet<> type here. Thus, return type would be this:

```ts
{
  readonly focusedRow: GetOrSet<Row | undefined>;
  readonly focusPath: GetOrSet<RA<number>>;
}
```

and the actual return statement like this:

```ts
return {
  focusRow: React.useState(undefined),
  focusPath: [focusPath, setFocusAndCachePath],
};
```

That should be a bit simpler and cleaner

There is no reason to expose both setFocusPath and setFocusAndCachePath to the outside - only setFocusAndCachePath should be exposed

---

though?